builtins/src/cost_modeling.rs

```diff
@@ -0,0 +1,162 @@
+//! Configurations for handling cost modeling of builtin programs.
+
+use {
+    crate::BUILTINS,
+    ahash::AHashMap,
+    lazy_static::lazy_static,
+    solana_feature_set::FeatureSet,
+    solana_pubkey::Pubkey,
+    solana_sdk_ids::{ed25519_program, secp256k1_program},
+};
+
+/// Configuration for cost modeling of a builtin program.
+#[derive(Debug)]
+pub struct CostModelingConfig {
+    /// The default cost of the builtin program.
+    /// If None, this builtin program is not cost modeled.
+    pub default_cost: Option<u64>,
+    /// Configuration for updating the cost of the builtin program.
+    /// If None, the cost is never updated.
+    pub new_cost_config: Option<NewCostModelingConfig>,
+}
+
+/// Configuration for updating the cost of a builtin program.
+#[derive(Clone, Copy, Debug)]
+pub struct NewCostModelingConfig {
+    /// The new cost of the builtin program.
+    pub new_cost: u64,
+    /// The feature gate to trigger the new cost.
+    pub feature_id: Pubkey,
+}
+
+struct BuiltinCost {
+    default_cost: Option<u64>,
+    core_bpf_migration_feature: Option<Pubkey>,
+    new_cost_config: Option<NewCostModelingConfig>,
+}
+
+lazy_static! {
+    static ref BUILTIN_INSTRUCTION_COSTS: AHashMap<Pubkey, BuiltinCost> = BUILTINS
+        .iter()
+        .map(|builtin| {
+            (
+                builtin.program_id,
+                BuiltinCost {
+                    default_cost: builtin.cost_modeling_config.default_cost,
+                    core_bpf_migration_feature: builtin
+                        .core_bpf_migration_config
+                        .as_ref()
+                        .map(|config| config.feature_id),
+                    new_cost_config: builtin.cost_modeling_config.new_cost_config,
+                },
+            )
+        })
+        .chain(
+            [
+                (
+                    secp256k1_program::id(),
```

Comment:
+1 for not forgetting these 2.

the new `secp256r1` probably will never be added to this list